server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -169,7 +170,14 @@ public void processFlushEvent(ShuffleDataFlushEvent event) throws Exception {
               storageDataReplica,
               user,
               maxConcurrencyPerPartitionToWrite);
-      ShuffleWriteHandler handler = storage.getOrCreateWriteHandler(request);
+      ShuffleWriteHandler handler;
+      try {
+        handler = storage.getOrCreateWriteHandler(request);
```

Comment:
If there is no write permission, the retry will overcome this case? Or just retry again to write this disk

Reply:
BTW, I hope this could be tested in the UTs

Reply:
> If there is no write permission, the retry will overcome this case? Or just retry again to write this disk

It dependency the logic of `StorageManagerSelector`. If EventRetryException be catch in `org.apache.uniffle.server.DefaultFlushEventHandler#handleEventAndUpdateMetrics`, the retryTimes of the flush event will be increase

Reply:
Got it. +1 for this